test/tmt/tests.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -exo pipefail
+
+if [[ $1 == '' ]]; then
+    echo -e "Usage: podman-tests.sh STREAM\nSTREAM can be upstream or downstream"
+    exit 1
+fi
+
+STREAM=$1
+
+# `rhel` macro exists on RHEL, CentOS Stream, and Fedora ELN
+# `centos` macro exists only on CentOS Stream
+CENTOS_VERSION=$(rpm --eval '%{?centos}')
+RHEL_VERSION=$(rpm --eval '%{?rhel}')
+
+# For upstream tests, we need to test with podman and other packages from the
+# podman-next copr. For downstream tests (bodhi, errata), we don't need any
+# additional setup
+if [[ "$STREAM" == "upstream" ]]; then
+    # Use CentOS Stream 10 copr target for RHEL-10 until EPEL 10 becomes
+    # available
+    # `rhel` macro exists on RHEL, CentOS Stream, and Fedora ELN
+    # `centos` macro exists only on CentOS Stream
+    if [[ -n $CENTOS_VERSION || $RHEL_VERSION -ge 10 ]]; then
+        dnf -y copr enable rhcontainerbot/podman-next centos-stream-"$CENTOS_VERSION"
+    else
+        dnf -y copr enable rhcontainerbot/podman-next
+    fi
+    echo "priority=5" >> /etc/yum.repos.d/_copr:copr.fedorainfracloud.org:rhcontainerbot:podman-next.repo
+fi
+
+# Remove testing-farm repos if they exist because they interfere with the
+# podman-next copr. The default distro repos will not be removed and can be
+# used wherever relevant.
+rm -f /etc/yum.repos.d/tag-repository.repo
+
+# Enable EPEL on RHEL/CentOS Stream envs to fetch bats
+if [[ -n $(rpm --eval '%{?rhel}') ]]; then
+    # Until EPEL 10 is available use epel-9 for all RHEL and CentOS Stream
+    dnf -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
+    sed -i 's/$releasever/9/g' /etc/yum.repos.d/epel.repo
+fi
+
+# Install dependencies for running tests
+dnf -y install bats bind-utils cargo clippy go-md2man iptables jq make netavark nftables nmap-ncat rustfmt slirp4netns
+
+cat /etc/redhat-release
```

Comment:
I think this output would be more useful to have at the start of the script.  If something above goes south, execution will never reach here.

Reply:
done